### PR TITLE
[Feat] Support mtaext files in mta read operations

### PR DESCRIPTION
**Pull request by tal-sapan**
The extensions are sent in parameter "-x" or "--extensions".
Currently, don't perform validations on the mtaext files except unmarshalling and during the merge itself. Use the partial result if errors occurs during the merge and return the errors in a "messages" property on the result.
When using mtaext files, the hash is not returned to ensure the write operations are performed on the data from the mta.yaml alone.

Also: move function readFile to file_ops and remove its duplications, and rename some variables used in commands.

### Checklist
- [X] Code compiles correctly
- [X] Relevant tests were added (unit / contract / integration)
- [X] Relevant logs were added
- [X] Formatting and linting run locally successfully
- [X] All tests passing
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: internal/resolver/mtaresolver.go [hunk 1]
```diff
@@ -35,17 +33,13 @@ type ResolveResult struct {
...
 	}
-	yamlData, err := ioutil.ReadFile(path)
+	mtaRaw, messages, err := mta.GetMtaFromFile(path, extensions)
 	if err != nil {
-		return result, errors.Wrapf(err, pathNotFoundMsg, path)
-	}
-	mtaRaw, err := mta.Unmarshal(yamlData)
-	if err != nil {
-		return result, errors.Wrapf(err, unmarshalFailsMsg, path)
+		return result, messages, errors.Wrapf(err, unmarshalFailsMsg, path)
```
Comment (allaVolkov) [inline internal/resolver/mtaresolver.go:42]: I think unmarshalFailsMsg message is not relevant now.

File: internal/resolver/mtaresolver.go [hunk 2]
```diff
@@ -65,15 +59,15 @@ func Resolve(workspaceDir, moduleName, path string, envFile string) (result Reso
 
 			propVarMap, err := getPropertiesAsEnvVar(module)
 			if err != nil {
-				return result, err
+				return result, messages, err
```
Comment (allaVolkov) [inline internal/resolver/mtaresolver.go:62]: don't we need result.Messages = m.messages here too? it's different from lines 65-66

Comment (tal-sapan) [reply internal/resolver/mtaresolver.go:62]: `result.Messages` is for the resolve result (like unresolved env variables). I want to have a separation in these messages so the interface will be consistent (the messages about bad mtaext files will be in the returned messages parameter for all commands).

File: internal/resolver/mtaresolver.go (hunk 1 above)
Comment (tal-sapan) [reply internal/resolver/mtaresolver.go]: The message is still relevant, I think - we want a good error message so the user will understand the issue. Without it, when unmarshal fails, the user will get messages like:
` yaml: unmarshal errors:\n  line 7: field type1 not found in type mta.Module`
I improved the message text and changed it message to be returned from `GetMtaFromFile` so we will always show it.

File: mta/mta_services.go [hunk 3]
```diff
@@ -83,71 +91,71 @@ func DeleteMta(path string) error {
...
+		return nil, messages, err
 	}
 	if len(workspaceDir) == 0 {
 		workspaceDir = filepath.Dir(path)
 	}
 
 	resource := mta.GetResourceByName(resourceName)
 	if resource == nil {
-		return nil, fmt.Errorf("the '%s' resource does not exist", resourceName)
+		return nil, messages, fmt.Errorf("the '%s' resource does not exist", resourceName)
```
Comment (allaVolkov) [inline mta/mta_services.go:158]: in most places we use like: `the "%s" resource does not exist`

File: mta/mta_services.go [hunk 4]
```diff
@@ -202,103 +210,103 @@ func mergeMaps(first map[string]interface{}, second map[string]interface{}) map[
...
 	for index, existingModule := range mtaObj.Modules {
 		if existingModule.Name == module.Name {
 			mtaObj.Modules[index] = &module
-			return saveMTA(path, mtaObj, marshal)
+			return messages, saveMTA(path, mtaObj, marshal)
 		}
 	}
 
-	return fmt.Errorf("the '%s' module does not exist", module.Name)
+	return messages, fmt.Errorf("the '%s' module does not exist", module.Name)
```
Comment (allaVolkov) [inline mta/mta_services.go:233]: not must, but you can fix message like in previous comment

**Review by allaVolkov — APPROVED** [review]

File: mta/mta_services.go (hunk 3 above)
Comment (tal-sapan) [reply mta/mta_services.go:158]: I didn't change the error message in this PR so I think I will leave it like this

**Pull request merged**